File: bpf/bcc/hello.py
#!/usr/bin/python
# ref: <Learning eBPF>
from bcc import BPF

program = r"""
int hello(void *ctx) {
    bpf_trace_printk("Hello World!");
    return 0;
}
"""

print("Tracing execve syscall ... Hit Ctrl-C to end")

b = BPF(text=program)
syscall = b.get_syscall_fnname("execve")
b.attach_kprobe(event=syscall, fn_name="hello")
b.trace_print()

File: cpp/hello.cpp
#include <stdio.h>
#include <limits.h>
#include <iostream>

class Hello {
	int print(void);
};

int Hello::print(void)
{
	std::cout << "Hello World!" << std::endl;
	return 0;
}

int main(void)
{
	std::cout << "Hello World!" << std::endl;
	return 0;
}
